jsonrpc/src/cancellation.ts

```diff
@@ -87,7 +87,13 @@ class MutableToken implements CancellationToken {
 	}
 }
 
-export class CancellationTokenSource {
+export abstract class AbstractCancellationTokenSource implements Disposable {
```

Comment:
would like to change to interface but not sure how.

Reply:
export interface AbstractCancellationTokenSource extends Disposable {
	token: CancellationToken;
	cancel(): void;
}

Reply:
so, you are fine with name? Abstract in interface seems weird. but if you are fine, I can change to interface.